haystack_experimental/components/extractors/llm_metadata_extractor.py

```diff
@@ -240,28 +253,80 @@ def from_dict(cls, data: Dict[str, Any]) -> "LLMMetadataExtractor":
             deserialize_secrets_inplace(data["init_parameters"]["generator_api_params"], keys=["api_key"])
         return default_from_dict(cls, data)
 
+    def _extract_metadata_and_update_doc(self, document: Document, errors: Dict[str, Any], content: str):
+        """
+        Extract metadata from the content and updates the document's metadata with the extracted metadata.
+
+        If the extraction fails, i.e.: no JSON is returned by the LLM API, the error message will be stored in
+        `errors`.
+
+        :param document: Document to be updated with the extracted metadata.
+        :param errors: Dictionary to store error messages if the extraction fails.
+        :param content: Content to extract metadata from.
+        """
+        prompt_with_doc = self.builder.run(
+            template=self.prompt,
+            template_variables={self.prompt_variable:content}
+        )
+        result = self.llm_provider.run(prompt=prompt_with_doc["prompt"])
+        llm_answer = result["replies"][0]
+        if self.is_valid_json_and_has_expected_keys(expected=self.expected_keys, received=llm_answer):
+            extracted_metadata = json.loads(llm_answer)
+            for k in self.expected_keys:
+                document.meta[k] = extracted_metadata[k]
+        else:
+            errors[document.id] = llm_answer
```

Comment:
I don't think this is needed now, but in the future also adding in the error thrown by the JSON loader would be helpful to understand why the `llm_answer` isn't working.